pkg/ssh/connection.go

```diff
@@ -134,7 +145,29 @@ func NewConnection(connector *Connector, opts Opts) (executor.Interface, error)
 			}
 		}
 
-		authMethods = append(authMethods, ssh.PublicKeys(signer))
+		if len(opts.SSHCert) > 0 {
+			cert, _, _, _, err := ssh.ParseAuthorizedKey([]byte(opts.SSHCert))
+			if err != nil {
+				return nil, fail.SSHError{
+					Op:  "parsing certificate",
+					Err: errors.Wrapf(err, "SSH certificate could not be parsed"),
+				}
+			}
+
+			// create a signer using both the certificate and the private key:
+			certSigner, signersErr := ssh.NewCertSigner(cert.(*ssh.Certificate), signer)
+			if err != nil {
```

Comment:
you're checking the wrong error here, should be

```suggestion
			if signersErr != nil {
```